renku/ui/cli/utils/callback.py

```diff
@@ -35,26 +35,43 @@ def __init__(self):
         super().__init__()
         self._progress_bars = {}
         self._progress_types = ["download"]
+        self._spinner = None
 
     def echo(self, msg, end="\n"):
         """Write a message."""
         with CommunicationCallback.lock:
+            if self._spinner:
+                self._spinner.hide()
```

Comment:
Why do we need to hide `_spinner` here? Also, it seems that we can use `hidden` context manager to temporarily hide the spinner if needed (just a matter of taste):
```
with self._spinner.hidden():
    print(...)
```

Reply:
We should also refactor all these methods to a single one.

Reply:
Spinner swallows all print output while active

Reply:
also using the `hidden` contextmanager doesn't work because _spinner might be None. I'll make a custom context manager